### ISSUE TITLE
Using S3 (DO Spaces) State Backend and Import 1 Resources Takes 5 Minutes

### ISSUE THREAD

**Issue by eproxus** [pulumi/pulumi#6888]
When using an external S3 backend (DigitalOcean Spaces) a creation of one stack and importing one DigitalOcean Project resource takes 5 minutes!

## Expected behavior
Creating a stack and importing one simple project resource should take less than one second (or at most a few seconds).

## Current behavior
It takes 5 minutes.
```
pulumi import digitalocean:index/project:Project websites e344f594-fa50-49ab-...

Previewing import (dev):
     Type                           Name                            Plan
 +   pulumi:pulumi:Stack            infrastructure-dev              create
 =   └─ digitalocean:index:Project  websites                        import

Resources:
    + 1 to create
    = 1 to import
    2 changes

Do you want to perform this import?  [Use arrows to move, enter to select, type Do you want to perform this import? yes
Importing (dev):
     Type                           Name                            Status
 +   pulumi:pulumi:Stack            infrastructure-dev              created   # This...
 =   └─ digitalocean:index:Project  websites                        importe   # ...and this takes ~5 minutes. Terminal keeps flickering

Resources:
    + 1 created
    = 1 imported
    2 changes

Duration: 5m9s

Please copy the following code into your Pulumi application. Not doing so
will cause Pulumi to report that an update will happen on the next update command.

Please note, that the imported resources are marked as protected. To destroy them
you will need to remove the `protect` option and run `pulumi update` *before*
the destroy will take effect.

import pulumi
import pulumi_digitalocean as digitalocean

websites = digitalocean.Project("websites",
    description="Public websites",
    environment="Production",
    name="websites",
    purpose="Website or blog",
    opts=pulumi.ResourceOptions(protect=True))
```

## Steps to reproduce
1. Create a new Pulumi project
2. Login to a "local" S3 backend using Digital Ocean spaces
3. Create a stack
4. Create a DigitalOcean Project in the DigitalOcean dashboard
5. Import the project to Pulumi
6. _Wait..._

## Context (Environment)

Total stack size as reported by DigitalOcean is 13 Kb in the end.

I'm a single developer in a small company. We can't (yet) justify using the Pulumi cloud and wanted to start with an S3-backed state to try things out. It is so slow it almost feels "intentionally" crippled.

## Affected feature

* `area/backends`
* #4605
* #6074

**Comment by stack72** [comment]
Hi @eproxus 

I'm sorry this experience is bad for you - I agree that it's slow. From our initial investigations here, we have seen that the DO Spaces API is actually rate limiting. There is nothing in our code path that cripples the backend storage performance depending on what vendor - we actually use https://github.com/google/go-cloud for interacting with backends

We can, of course, investigate this more but this your findings replicates exactly what I've seen before.

Paul

**Comment by eproxus** [comment]
How many requests are you actually generated for it to hit the rate limiter? It seems excessive either way (see also #6074)...

**Comment by eproxus** [comment]
@stack72 I did try Amazon S3 and the difference is night and day. Not sure what DO are actually doing, but it seems their Spaces is totally unusable for this use case (and _many/all_ other cases I assume!).

**Comment by stack72** [comment]
@eproxus yeah, I have tests for AliCloud / Minio / Ceph / S3 and Spaces and only spaces actually shows performance problems :/

I agree with you on the issue linked above and that definitely causes some of the problems and we are going to be looking into that issue soon!

I'm glad you are able to see the same sort of performance as I am seeing
